github/resource_github_branch_protection.go

```diff
@@ -269,3 +269,24 @@ func resourceGithubBranchProtectionDelete(d *schema.ResourceData, meta interface
 
 	return err
 }
+
+func resourceGithubBranchProtectionImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	repoName, pattern, err := parseTwoPartID(d.Id(), "repository", "pattern")
```

Comment:
What happens if someone passes in the `nodeid` of the branch protection rule here? Will that functionality still work? ex. is the plan to support both of these commands?

```
terraform import github_branch_protection.test MDIwOkJyYW5jaFByb3RlY3Rpb25SdWxlMTgyNDY3NDE=
```

```
terraform import github_branch_protection.test branch-protection-test:main
```

Reply:
Nice catch. I started down the path of supporting both options it ended up sticking to what is documented. Passing node ID will fail.